pulser-core/pulser/register/register.py

```diff
@@ -297,6 +310,33 @@ def rotate(self, degrees: float) -> None:
         )
         object.__setattr__(self, "_coords", [rot @ v for v in self._coords])
 
+    def rotated(self, degrees: float) -> Register:
+        """Makes a new rotated register.
+
+        All coordinates are rotated counter-clockwise around the origin.
+
+        Args:
+            degrees: The angle of rotation in degrees.
+
+        Returns:
+            Register: A new register rotated around the origin by the given
+            angle.
+        """
+        theta = np.deg2rad(degrees)
+        rot = np.array(
+            [[np.cos(theta), -np.sin(theta)], [np.sin(theta), np.cos(theta)]]
+        )
+        if self.layout is not None:
+            warnings.warn(
+                "The rotated register won't have an associated "
+                "'RegisterLayout'.",
```

Comment:
Perhaps specify the category ? It's a `Userwarning` by default and it might never change

Reply:
I also feel like there is a way to define the rotation for both layout and register, by defining a method `get_rotated_coords(self, angle)` in CoordsCollection (that would only work for 2D coords/rotate around phi=0 in 3D). But I guess it's not very interesting to rotate a register layout since it is obtained from calibration ?

Reply:
I doubt `warnings.warn()` will ever change the default warning category in Python 3.x, but I can still specify it. 
As you suggested, register layouts should give a notion of being static so having an easy option to rotate them is not something we want to be giving the user imo

Reply:
Yes let's go with this PR !